usr.sbin/kldxref/kldxref.c

```diff
@@ -717,13 +717,13 @@ read_kld(char *filename, char *kldname)
 static FILE *
 maketempfile(char *dest, const char *root)
 {
-	char *p;
-	int n, fd;
+	struct stat stroot;
+	int fd;
 
-	p = strrchr(root, '/');
-	n = p != NULL ? p - root + 1 : 0;
-	if (snprintf(dest, MAXPATHLEN, "%.*slhint.XXXXXX", n, root) >=
-	    MAXPATHLEN) {
+	if (stat(root, &stroot) != 0 || !S_ISDIR(stroot.st_mode))
+		return (NULL);
+	if (snprintf(dest, MAXPATHLEN, "%.*s/lhint.XXXXXX", (int)strlen(root),
+	    root) >= MAXPATHLEN) {
```

Comment:
```suggestion
	if (snprintf(dest, MAXPATHLEN, "%s/lhint.XXXXXX", root) >= MAXPATHLEN) {
```

Subject to line wrapping.